### PR TITLE
ppgbp modifications

### PR DESCRIPTION
**Pull request by Francesco-Carlucci**
Adding all modifications needed to support the ppgbp architectures, ResNet and UNet like.

1.  getitem operation support
    - adding 'features_slicing' in meta
    - adding correct_get_item to fix indices during exports
    - adding GetItemFeaturesCalculator
    - tests on group conv with any number of groups
    - test with any possible indexing method (slice, list, integer, boolean mask and torch tensor
2. Adding PITInstanceNorm and PITPReLU classes
3. adding minor propagting layers (sigmoid, upsample...)
4.  fix group conv in supernet, using index instead of name
5. adding None value in ModAttrFeaturesCalculator, so that GetItemFeaturesCalculator works with MPS, where there are no masks
6. adding group convolution notebook in examples/

### CODE REVIEW

File: plinio/graph/inspection.py [hunk 1]
```diff
@@ -433,6 +463,46 @@ def is_concatenate(n: fx.Node, parent: fx.GraphModule) -> bool:
...
+        #check if length of padding exceed the [H] [D] W dimensions of the input vector
+        if floor(len(pad)/2) > len(n.all_input_nodes[0].meta['tensor_meta'].shape)-2: #subtract batch_size, #channels
+            return True
+        #is_features_dim = slices[1] != slice(None,None,None)
+        #if is_features_dim:
+        #    return True
+    elif n.op == 'call_module':
+        submodule = parent.get_submodule(str(n.target))
+        if isinstance(submodule, nn.ConstantPad1d): #if floor(len(pad)/2) > len(n.all_input_nodes[0].meta['tensor_meta'].shape)-2:
+            return floor(len(submodule.padding)/2) > len(n.all_input_nodes[0].meta['tensor_meta'].shape)-2
```
[the remark this answers is not in the document]
Comment (danielepagliari) [reply plinio/graph/inspection.py]: Not sure I understood this. See my comment below.

File: unit_test/test_methods/test_mps/test_mps_convert.py [hunk 2]
```diff
@@ -28,11 +28,15 @@
 import plinio.methods.mps.quant.nn as qnn
 from plinio.methods.mps.quant.quantizers import FQWeight, MinMaxWeight, DummyQuantizer, PACTAct
 from unit_test.models import SimpleNN, SimpleNN2D, DSCNN, ToyMultiPath1_2D, ToyAdd_2D, \
-    SimpleMPSNN, SimpleExportedNN1D, SimpleExportedNN2D, SimpleExportedNN2D_ch, SimpleNN2D_NoBN
+    SimpleMPSNN, SimpleExportedNN1D, SimpleExportedNN2D, SimpleExportedNN2D_ch, SimpleNN2D_NoBN, \
+    ToyGroupedConv_1D
```
[the remark this answers is not in the document]
Comment (danielepagliari) [reply unit_test/test_methods/test_mps/test_mps_convert.py:32]: Ok, let's try to support PER_CHANNEL too asap.

File: plinio/methods/pit/nn/conv1d.py [hunk 3]
```diff
@@ -223,7 +223,15 @@ def export(n: fx.Node, mod: fx.GraphModule):
...
-            new_bn = nn.BatchNorm1d(
+            if isinstance(submodule.bn, nn.BatchNorm1d):
+                norm_layer = nn.BatchNorm1d
+            elif isinstance(submodule.bn, nn.BatchNorm2d):
+                norm_layer = nn.BatchNorm2d
+            elif isinstance(submodule.bn, nn.InstanceNorm1d):
+                norm_layer = nn.InstanceNorm1d
+            else:
+                raise ValueError(f"Unsupported normalizer type {type(submodule.bn)} for PITConv1d export")
+            new_bn = norm_layer(
```
Comment (danielepagliari) [inline plinio/methods/pit/nn/conv1d.py:234]: Really last comment:
- BatchNorm2d should not happen after Conv1d, right?
- Can't we make it more general using simply: norm_layer = type(submodule.bn)?

**Review by danielepagliari — APPROVED** [review]
I left a final super-minor comment, but the PR can be accepted.

File: plinio/methods/pit/nn/conv1d.py (hunk 3 above)
Comment (danielepagliari) [reply plinio/methods/pit/nn/conv1d.py:234]: Also, why didn't you add similar code for Conv2d and Conv3d?
